src/index.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+// const temperature = 1  // for now CHANGE LATER
+// const state = {
+let tempIncreaseButton = null;
+let tempDecreaseButton = null;
+let getRealTimeTemp = null;
+let tempValue = null;
+let gardenContent = null;
+let temperature = 79;
+let headerCity = null;
+let cityNameUserInput = null;
+let defaultCityName = 'Princeton';
+let skyTemplate = null;
+let skyOption = null;
+let skyInput = null;
+let resetCityName = null;
+let realTimeTemperature = null;
+// }
+
+const selectTag = () => {
+    tempIncreaseButton = document.getElementById('increaseTempControl');
+    tempDecreaseButton = document.getElementById('decreaseTempControl');
+    tempValue = document.getElementById('tempValue');
+    gardenContent = document.getElementById('landscape');
+    headerCity = document.getElementById('headerCityName');
+    cityNameUserInput = document.getElementById('cityNameInput');
+    skyInput = document.getElementById('skySelect');
+    skyTemplate = document.getElementById('sky');
+    resetCityName = document.getElementById('cityNameReset');
+    realTimeTemperature = document.getElementById('currentTempButton');
+};
+//Create optionsa element for sky in the HTML file
+const createOptions = () => {
+    const optionArray = ['Sunny', 'Cloudy', 'Snowy', 'Rainy'];
+    for (let i = 0; i < optionArray.length; ++i) {
+        skyOption = document.createElement('option');
+        skyOption.textContent = optionArray[i];
+        skyInput.appendChild(skyOption);
+    }
+};
+
+const registerEvent = () => {
+    tempIncreaseButton.addEventListener('click', handleIncreaseTempClick);
+    tempDecreaseButton.addEventListener('click', handleDecreaseTempClick);
+    resetCityName.addEventListener('click', handleResetButton);
+    realTimeTemperature.addEventListener('click', handleRealTemp);
+    skyInput.addEventListener('change', skyInputHandler);
+    cityNameUserInput.addEventListener('input', headerCityFunction);
+};
+
+const handleIncreaseTempClick = () => {
+    ++temperature;
+    showTemp();
+};
+
+const handleDecreaseTempClick = () => {
+    --temperature;
+    showTemp();
+};
+
+
+
+const showTemp = () => {
+    tempValue.textContent = temperature;
+    updateTempColor();
+};
+
+const headerCityFunction = () => {
+    headerCity.textContent = cityNameUserInput.value; //defaultCityName;
+};
+
+const skyInputHandler = () => {
+    const input = skyInput.value;
+    skyTemplate.textContent = sky[input];
+};
+
+const handleResetButton = () => {
+    cityNameUserInput.value = '';
+    headerCity.textContent = '';
+};
+
+// wave 4 - calling API
+const handleRealTemp = async () => {
+    tempValue.textContent = Math.floor((await getWeather() - 273.15) * 9/5 + 32 );
```

Comment:
To make this line really clear, I'd suggest splitting up fetching the temperature from converting it to Fahrenheit.